server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -265,6 +269,16 @@ protected final void innerWriteTo(StreamOutput out, boolean asKey) throws IOExce
             out.writeOptionalWriteable(readerId);
             out.writeOptionalTimeValue(keepAlive);
         }
+
+        // Do not serialise cacheModifier if it is not used for key. If cacheModifier is
+        // needed for a node, the interceptor should do the right job of setting it.
+        // If we allow it to be passed across the wire, the search result on the remote node
+        // may get surprising result, e.g. no dls/fls is applicable to the remote node, but
+        // the search result is limited because the cacheModifier makes it match a limited
+        // result in the cache.
+        if (asKey && out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeOptionalBytesReference(cacheModifier);
+        }
```

Comment:
@tvernum I changed this to not serialise across wire for the reason detailed in the comment.